dataflow/src/main/java/org/checkerframework/dataflow/analysis/AbstractAnalysis.java

```diff
@@ -307,7 +307,9 @@ public Direction getDirection() {
     }
 
     /**
-     * Call the transfer function for node {@code node}, and set that node as current node first.
+     * Call the transfer function for node {@code node}, and set that node as current node first. Be
+     * careful that {@code store} may be shared between nodes in a block. Passing a copied store to
+     * avoid the accident changing.
```

Comment:
I don't fully understand this comment. I think it means that `store` should not be side-effected, because it is a copy of a store used elsewhere. If that's correct, can you change the documentation to say it directly.

Also, I find `store` to be a misleading name, because I would expect it to be a `Store`, but it is a `TransferInput`. Should we change the name?

Reply:
What I intend to say here is that we should remember to pass a copied `store` when calling this method, otherwise the node's transfer input may be changed in the future. Here is an example: https://github.com/typetools/checker-framework/pull/3370/files#diff-9f052b8cabea4817c69534ba908adc75R349. The node in the next iteration maybe change the transfer input of the previous node. Could you please give me some advice?

`store` does not mean `Store` here, one parameter's name of this method is `store`. I can make it clear in Javadoc.

Reply:
> store does not mean Store here, one parameter's name of this method is store. I can make it clear in Javadoc.

Martin suggested that you rename the formal parameter.  It is better to give the variable a meaningful name, rather than retain a bad name and add documentation to explain how it is bad.

Reply:
Most of the code in `AbstractAnalysis` is what previously was in `Analysis`.
https://github.com/typetools/checker-framework/blob/master/dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java#L427
Instead of just locally renaming this parameter, we can perform a clean-up later and make sure there are no `TransferInput` parameters with name `store` and that also the arguments have appropriate names.